oauthproxy.go

```diff
@@ -480,7 +486,9 @@ func (p *OAuthProxy) IsValidRedirect(redirect string) bool {
 }
 
 func (p *OAuthProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	p.preAuthChain.Then(http.HandlerFunc(p.serveHTTP)).ServeHTTP(rw, req)
+	p.preAuthChain.
+		Then(http.HandlerFunc(p.serveHTTP)).
+		ServeHTTP(middleware.NewResponseWriter(rw), req)
```

Comment:
Do we want our custom ResponseWriter that tracks size &  status code available for all middleware? (like this does).

Or only have middleware that needs it wrap a ResponseWriter (for now just the logging middleware seems to need this).

Reply:
I guess the only thing with having others is that it's a bit random, and you can't use the logging middleware without being aware of it.

In terms of encapsulation of the logic of the software I would tie this all together and have it be explicitly part of the logging middleware. That way you can forget to use it and break the logging.

If another piece of the project needs a response hijacker they can create their own for their own purposes

Reply:
Yeah, that makes sense to me. I was torn on the right direction -- I think I saw this pattern over Christmas break looking at frameworks as a guide.

But we have much more power over our middleware and each one can make there own that has the customizations the need on the `http.ResponseWriter`